operators/pkg/controller/elasticsearch/sset/list.go

```diff
@@ -61,6 +67,24 @@ func (l StatefulSetList) PodNames() []string {
 	return names
 }
 
+// GetActualPods returns the list of pods currently existing in the StatefulSetList.
+// Some "expected" pods may be missing from the returned list if they don't exist yet
+// or are temporarily down.
+func (l StatefulSetList) GetActualPods(c k8s.Client) ([]corev1.Pod, error) {
+	if len(l) == 0 {
+		return nil, nil
+	}
+	allPods := make([]corev1.Pod, 0, len(l))
```

Comment:
capacity of len(l) is almost arbitrary, and might as well be skipped? 1 replica per statefulset is not a case worth optimizing for, is it?